packages/datagateway-search/src/searchPageContainer.component.tsx

```diff
@@ -43,6 +46,79 @@ import ViewListIcon from '@material-ui/icons/ViewList';
 import ViewAgendaIcon from '@material-ui/icons/ViewAgenda';
 import { StyleRules } from '@material-ui/core/styles';
 
+export const storeFilters = (
+  filters: FiltersType,
+  searchableEntities: string
+): void => {
+  const filter = (searchableEntities as string) + 'Filters';
+
+  localStorage.setItem(filter, JSON.stringify(filters));
+};
+
+export const storeSort = (
+  sorts: SortType,
+  searchableEntities: string
+): void => {
+  const sort = (searchableEntities as string) + 'Sort';
+
+  localStorage.setItem(sort, JSON.stringify(sorts));
+};
+
+export const storePage = (page: number, searchableEntities: string): void => {
+  const pageNumber = (searchableEntities as string) + 'Page';
+
+  localStorage.setItem(pageNumber, JSON.stringify(page));
+};
+
+export const storeResults = (
+  results: number,
+  searchableEntities: string
+): void => {
+  const resultsNumber = (searchableEntities as string) + 'Results';
+
+  localStorage.setItem(resultsNumber, JSON.stringify(results));
+};
```

Comment:
Minor optimisation - a lot of the time these will end up storing the default value in `localStorage` - and thus cause us to update the query params later on as the result from localStorage is not null. Perhaps some guards to prevent this will optimise updates a little?

i.e.
```
if (Object.keys(filters).length !== 0)
   localStorage.setItem(filter, JSON.stringify(filters));
```
and
```
if (page !== 1)
   localStorage.setItem(filter, JSON.stringify(page));
```